src/lib/Microsoft.Health.Events/Telemetry/Exceptions/EventHubExceptionTelemetryProcessor.cs

```diff
@@ -23,26 +23,69 @@ public static class EventHubExceptionTelemetryProcessor
             EnsureArg.IsNotNull(exception, nameof(exception));
             EnsureArg.IsNotNull(logger, nameof(logger));
 
-            logger.LogError(exception);
+            var ex = CustomizeException(exception);
+
+            logger.LogError(ex);
 
             if (shouldLogMetric)
             {
-                logger.LogMetric(EventMetrics.HandledException(errorMetricName ?? GetErrorMetricName(exception), ConnectorOperation.Setup), 1);
+                if (ex.Equals(exception))
+                {
+                    logger.LogMetric(
+                        EventMetrics.HandledException(
+                            errorMetricName ?? $"{ErrorType.EventHubError}{EventHubErrorCode.GeneralError}",
+                            ConnectorOperation.Setup),
+                        1);
+                }
+                else
+                {
+                    var processor = new EventHubConfigurationExceptionTelemetryProcessor();
+                    processor.HandleException(ex, logger, ConnectorOperation.Setup);
+                }
             }
         }
 
-        private static string GetErrorMetricName(Exception exception)
+        public static Exception CustomizeException(Exception exception)
         {
+            string message = exception.Message;
+            string helpLink = exception.HelpLink;
+
             switch (exception)
             {
                 case EventHubsException _:
-                    return $"{ErrorType.EventHubError}{EventHubErrorCode.OperationError}";
+                    EventHubsException eventHubsException = (EventHubsException)exception;
+                    switch (eventHubsException.Reason)
+                    {
+                        case EventHubsException.FailureReason.ConsumerDisconnected:
+                            message = "Verify that the provided Event Hub is not already receiving data for another IoT connector or Azure resource.";
```

Comment:
**Background information:**

The `ConsumerDisconnected` reason means that "A consumer client was disconnected by the Event Hub service from the Event Hub instance. It typically occurs when a consumer with a higher owner level asserts ownership over a partition and consumer group pairing."

I encountered the `EventHubsException` exception and `ConsumerDisconnected` reason only when testing locally with a ConnectionString. To trigger this exception and reason, I pointed to another Event Hub with the following changes to the InputEventHub:ConnectionString:
- Set `Endpoint` to a new Namespace's FQDN in a new Resource Group (which was deployed via Sandbox)
- Ensured this new Namespace I'm pointing at contains an Event Hub with the same SAS policy as the previous Event Hub's claims (it was the Manage claim which includes the Listen claim), and then updated `SharedAccessKeyName` and `SharedAccessKey` to this new SAS policy

The exception message was `Receiver '<an id>' with a higher epoch '1' already exists. Receiver 'nil' with epoch 0 cannot be created. Make sure you are creating receiver with increasing epoch value to ensure connectivity, or ensure all old epoch receivers are closed or disconnected.`

**Thoughts:**

Given this exception message and that the new Event Hub I tested with was deployed via Sandbox, I thought that the exception was triggered if the Event Hub is already receiving data for another resource -- which, now that you point it out and I re-think it, doesn't make sense since there can be data from multiple devices (my test here was for Normalization).
Would this custom exception message be better?: _"Verify that the provided Event Hub's **consumer group** is not already receiving data for another IoT connector or Azure resource."_
Please correct me if I'm wrong, but it's technically the "partition" that shouldn't already be receiving data, but the user can't check/update the partition? So we can generalize with "consumer group" instead?

Reply:
Just to clarify, when you encountered this error during testing, did you have multiple consumers attached to the EventHub? The exception _sounds_ as though it's saying a new consumer has been added to the consumer group. If that is the case I'm wondering if this exception is expected; the EventHub client has logic to balance the load among multiple consumers.

Reply:
**Rephrasing background information:**
A receiver (consumer) hits an `EventHubsException` with `FailureReason.ConsumerDisconnected` if it can't read from a partition due to another receiver with a set higher "owner level" being the exclusive receiver/reader of the partition for the consumer group. (The owner level is a value to indicate a receiver's relative priority in being allowed to be the exclusive receiver.)

**Encountering `FailureReason.ConsumerDisconnected` when testing locally:** (Thanks, pallar-ms, for helping me understand this!)
I deployed a Gen1 test environment via [Sandbox](https://github.com/microsoft/iomt-fhir/blob/master/docs/Sandbox.md). Then, in my `appsettings.json`, I modified only the "InputEventHub:ConnectionString" to point at the devicedata Event Hub in the Gen1 test environment's Event Hubs Namespace (and updated the SharedAccessKey accordingly). However, since I ran my Console app locally, it acted as a receiver (a key point I missed before) and thus was competing with the Gen1 Function App as a receiver for the same devicedata Event Hub. I hit `FailureReason.ConsumerDisconnected` since my local app lost the owner level battle and couldn't draw events. Therefore, in contrast, once I "Stop" the Function App via Portal, I do not hit the `FailureReason.ConsumerDisconnected` anymore. (Initially I was also trying to understand why, if I specify a new consumer group in `appsettings.json`'s "InputEventHub:EventHubConsumerGroup", I was not hitting the `FailureReason.ConsumerDisconnected` - well, my local app was not competing with other consumers.)

**Testing with Gen2 test environment:**
pallar-ms and I tried to reproduce the `FailureReason.ConsumerDisconnected` in a somewhat similar scenario by pointing two IoT connectors deployed to read from the same devicedata Event Hub (by pointing the normalization pods to the same devicedata Event Hub), but we did not encounter it.

**Result:**
Since we have encountered the `FailureReason.ConsumerDisconnected` only with a local app and Gen1 Function App running at the same time and pointing to the same Event Hub, which is not a realistic scenario in production, I removed the `FailureReason.ConsumerDisconnected` case in my last commit of this PR. If we encounter it in the future we can observe how it was triggered and customize the exception then.